packages/fern-typescript/client/src/http/utils.ts

```diff
@@ -4,30 +4,35 @@ export function doesServiceHaveHeaders(service: HttpService): boolean {
     return service.headers.length > 0;
 }
 
-export function doesServiceHaveAuth(
+export function doesServiceHaveBearerAuth(
     service: HttpService
-): { hasAuth: false } | { hasAuth: true; isOptional: boolean; authType: HttpAuth.Bearer | HttpAuth.Basic } {
+): { hasAuth: false } | { hasAuth: true; isOptional: boolean } {
+    return doesServiceHaveAuth(service, HttpAuth.Bearer);
+}
+
+export function doesServiceHaveBasicAuth(
```

Comment:
having the 2 functions `doesServiceHaveBasicAuth` and `doesServiceHaveBearerAuth` is def cleaner cause it gets rid of the weird error case -- didn't do it originally cause then it would mean another iteration over the endpoints